pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -111,7 +111,10 @@ message MessageMetadata {
 	optional bytes encryption_param = 15;
 	optional bytes schema_version = 16;
 
-        optional bool partition_key_b64_encoded = 17 [ default = false ];
+	optional bool partition_key_b64_encoded = 17 [ default = false ];
+
+	// Mark the message to be delivered at or after the specified timestamp
+	optional uint64 deliver_at_time = 18;
```

Comment:
I would prefer using `int64` rather than `uint64` because Java doesn't have a unsigned int64.

Reply:
yes,  we can change this. I was just following the same time used for `publish_time` and `event_time`. In practical terms it shouldn't pose a problem either way.

Reply:
Switched to int64